docs/development/architecture.md

```diff
@@ -0,0 +1,180 @@
+- **backend/**: A Go-based server that communicates with the Kubernetes API and serves data to the frontend.
+- **app/**: A native Electron-based application for running Headlamp as a desktop app.
+- **plugins/headlamp-plugin**: A component to manage plugins in development and in deployment.
+- **plugins/examples**: Examples of how to achieve.
+- **docs/**: Documentation that is in markdown, and gets put on the Headlamp Website.
+
+The following components are in separate github repos:
+- **Plugins**: Extensible modules that add custom functionality to the UI. The ones maintained by the Headlamp team are in the [headlamp-k8s/plugins repo](https://github.com/headlamp-k8s/plugins). These include plugins for projects like Flux, Backstage and Inspektor Gadget.
+- **Headlamp Website**: Maintained in the [headlamp-k8s/website repo](https://github.com/headlamp-k8s/website). This contains things like the blog and the documentation. The website can be found at https://headlamp.dev/
+
+### Makefile task entry point
+
+The headlamp/ repo [Makefile](https://github.com/headlamp-k8s/headlamp/blob/main/Makefile) contains a number of targets for building and testing different components.
+
+### Frontend
+
+- Entry point is [frontend/src/index.tsx](https://github.com/headlamp-k8s/headlamp/blob/main/frontend/src/index.tsx)
+- See the [Frontend Documentation](https://headlamp.dev/docs/latest/development/frontend) for more information including a Quickstart.
+
+A few key dependencies:
+
+- **TypeScript**: Written in TypeScript.
+- **React Components**: Modular and reusable components that form the UI.
+- **Mui**: For UI components it uses Mui a Material UI component library.
+- **State Management**: Uses Redux for managing application state.
+- **Routing**: Utilizes React Router for navigation within the app.
+- **Npm**: Uses npm for package management.
+
+For each component we have a .stories.tsx file. For example Tables.stories.tsx for the Table.tsx component.
+
+### Backend (headlamp-server)
+
+- The [backend/](https://github.com/headlamp-k8s/headlamp/blob/main/backend/) folder contains the source code for the headlamp-server binary.
+- Entry point is [backend/cmd/headlamp.go](https://github.com/headlamp-k8s/headlamp/blob/main/backend/cmd/headlamp.go)
+- See the [Backend Documentation](https://headlamp.dev/docs/latest/development/backend/) for more information including a Quickstart.
+
+A few key dependencies:
+- **Golang**: Written using the [golang]() language. 
+- **Gorilla**: is the web framework we use
+- **Kubernetes and Helm**: Uses Kubernetes and Helm golang packages.
+
+Packages are stored in the [backend/pkg](https://github.com/headlamp-k8s/headlamp/blob/main/backend/pkg/) folder. The packages available are cache, config, helm, kubeconfig, logger, plugins, portforward, and utils. Future code should be organized in packages. There is still some code inside cmd/ folder that probably should be migrated to separate packages.
```

Comment:
I would not list the actual packages, since then we need to keep this up to date.